x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolverTests.java

```diff
@@ -2175,7 +2182,6 @@ public void testDataStreamNotAuthorizedWhenBackingIndicesAreAuthorizedViaWildcar
         assertThat(authorizedIndices.check("logs-foobar"), is(false));
         DataStream dataStream = metadata.dataStreams().get("logs-foobar");
         assertThat(authorizedIndices.all().get(), not(hasItem(indexName)));
-        assertThat(authorizedIndices.check(indexName), is(false));
```

Comment:
This is an interesting point. The indexName `.ds-logs-foobar-*` contains a wildcard. I think this `check` will now return `true` here since the new name check should make it pass. It previously returns `false` because the name does not have an IndexAbstraction. If we believe the new behaviour is correct (returning `true`), should we retain this assertion but change it to `is(true)` and add a comment to say why it is the case? Just deleting it feels like we are leaving it unspecified and I don't think we want that.

Reply:
IMO It doesn't make sense to test *patterns* at all for either `AuthorizedIndices#all` or `AuthorizedIndices#check`. In the main code it's only names of resources that are passed as arguments to these methods, never patterns. That's why I removed the test here. The assert can be `true` or `false` depending if the pattern in the request is greater than the the pattern in the permission, but it's not relevant.

Reply:
pushed [b8cd97e](https://github.com/elastic/elasticsearch/pull/92625/commits/b8cd97e51294304dcec8e0734e4adf209d9f20e8)

Reply:
> In the main code it's only names of resources that are passed as arguments to these methods, never patterns. 

I think it could happen when `expand_wildcards=none`. Not something we should optimize for. But the `check` should still work correctly for it.

Reply:
Fair enough, I forgot about that case.